### PR TITLE
Handled 400 error while fetching ACLs from bucket that has uniform level access

### LINKED ISSUE #3649: Handle Exceptions to Get Bucket ACLs when Uniform Bucket-Level Access is Enabled
## Story
When uniform-bucket-level-access to all buckets is enabled, Forseti will throw the following error when retrieving the legacy bucket ACLs.  This error is [supported by the documentation here](https://cloud.google.com/storage/docs/uniform-bucket-level-access).

Error with Uniform Bucket-Level Access (must add `Viewer` role to surface):
```
2020-02-14 12:34:11,910 ERROR [forseti-security][2.24.0] google.cloud.forseti.common.gcp_api.storage(get_bucket_acls): GCP API Error: unable to get bucketAccessControls, bucket = <redacted_bucket_name>, from GCP: <HttpError 400 when requesting https://storage.googleapis.com/storage/v1/b/<redacted_bucket_name>/acl?alt=json returned "Cannot get legacy ACL for a bucket that has uniform bucket-level access. Read more at https://cloud.google.com/storage/docs/uniform-bucket-level-access.">
```

Original Error (without `Viewer` role):
```
2020-02-13 10:06:37,654 ERROR [forseti-security][2.24.0] google.cloud.forseti.common.gcp_api.storage(get_bucket_acls): GCP API Error: unable to get bucketAccessControls, bucket = <redacted_bucket_name>, from GCP:

<HttpError 403 when requesting https://www.googleapis.com/storage/v1/b/<redacted_bucket_name>/acl?alt=json returned "<redacted_service_account> does not have storage.buckets.get access to <redacted_bucket_name>.">

{

  "error": {

    "code": 403,

    "message": "<redacted_service_account> does not have storage.buckets.get access to <redacted_bucket_name>.",

    "errors": [

      {

        "message": "<redacted_service_account> does not have storage.buckets.get access to <redacted_bucket_name>.",

        "domain": "global",

        "reason": "forbidden"

      }

    ]

  }

}

Traceback (most recent call last):

  File "/opt/forseti/.local/lib/python3.7/site-packages/forseti_security-2.24.0-py3.7.egg/google/cloud/forseti/common/gcp_api/storage.py", line 539, in get_bucket_acls
```

## Troubleshooting:
Reproduced by customer via offline email.

## Solution
Handle the exception, and make it log as info.   Or check whether Uniform Bucket-Level Access is enabled.

### PR DESCRIPTION
**Pull request by red2k18**
Resolves #3649 

Handled 400 error while fetching ACLs from bucket that has uniform level access.

### CODE REVIEW

File: google/cloud/forseti/common/gcp_api/storage.py [hunk 1]
```diff
@@ -562,6 +562,14 @@ def get_bucket_acls(self, bucket, user_project=None):
                                 'retrying.', bucket)
                     return self.get_bucket_acls(bucket, self._user_project)
 
+            raw_error = e.args[1]
+            error = raw_error.decode('utf-8')
+            formatted_error = json.loads(error)
+            error_code = formatted_error['error']['code']
+            if error_code == 400:
```
Comment (gkowalski-google) [inline google/cloud/forseti/common/gcp_api/storage.py]: Checking for 400 would only handle the case where the Forseti server service account has the Viewer role, but not case where the SA does not have that role. Can we try to instead not make the call if the storage object has uniform bucket level access enabled? It is part of the [CAI/API resource](https://cloud.google.com/storage/docs/json_api/v1/buckets#resource-representations).

**Comment by blueandgold** [comment]
@red2k18 is still working on this, trying to determine the best place to check the uniform bucket level access.

**Comment by red2k18** [comment]
@blueandgold , will add this to my 20% GitHub backlog work to the Sprint.

**Comment by red2k18** [comment]
This issue was not prioritized in the previous Sprint, and is not prioritized in the current Sprint based on the decisions made in the Sprint planning meet. Closing this PR as it will not be worked on in the current Sprint. Will reopen it when it is prioritized.

**Pull request closed without merging**